evap/staff/importers/json.py

```diff
@@ -0,0 +1,204 @@
+        self.course_type_cache: dict[str, CourseType] = {}
+        self.degree_cache: dict[str, Degree] = {}
+        self.course_map: dict[str, Course] = {}
+
+    def _get_course_type(self, name: str) -> CourseType:
+        if name in self.course_type_cache:
+            return self.course_type_cache[name]
+
+        course_type = CourseType.objects.get_or_create(name_de=name, defaults={"name_en": name})[0]
+        self.course_type_cache[name] = course_type
+        return course_type
+
+    def _get_degree(self, name: str) -> Degree:
+        if name in self.degree_cache:
+            return self.degree_cache[name]
+
+        degree = Degree.objects.get_or_create(name_de=name, defaults={"name_en": name})[0]
+        self.degree_cache[name] = degree
+        return degree
+
+    def _get_user_profiles(self, data: list[ImportRelated]) -> list[UserProfile]:
+        return [self.user_profile_map[related["gguid"]] for related in data]
+
+    def _import_students(self, data: list[ImportStudent]):
+        for entry in data:
+            email = clean_email(entry["email"])
+            user_profile, __ = UserProfile.objects.update_or_create(
+                email=email,
+                defaults={"last_name": entry["name"], "first_name_given": entry["christianname"]},
+            )
+
+            self.user_profile_map[entry["gguid"]] = user_profile
+
+    def _import_lecturers(self, data: list[ImportLecturer]):
+        for entry in data:
+            email = clean_email(entry["email"])
+            user_profile, __ = UserProfile.objects.update_or_create(
+                email=email,
+                defaults={
+                    "last_name": entry["name"],
+                    "first_name_given": entry["christianname"],
+                    "title": entry["titlefront"],
+                },
+            )
+
+            self.user_profile_map[entry["gguid"]] = user_profile
+
+    def _import_course(self, data: ImportEvent) -> Course:
+        course_type = self._get_course_type(data["type"])
+        degrees = [self._get_degree(c["cprid"]) for c in data["courses"]]
+        responsibles = self._get_user_profiles(data["lecturers"])
+        course, __ = Course.objects.update_or_create(
+            semester=self.semester,
+            cms_id=data["gguid"],
+            defaults={"name_de": data["title"], "name_en": data["title_en"], "type": course_type},
+        )
+        course.degrees.set(degrees)
+        course.responsibles.set(responsibles)
+
+        self.course_map[data["gguid"]] = course
+
+        return course
+
+    def _import_evaluation(self, course: Course, data: ImportEvent) -> Evaluation:
+        course_end = datetime.strptime(data["appointments"][0]["end"], self.DATETIME_FORMAT)
+
+        if data["isexam"]:
+            # Set evaluation time frame of three days for exam evaluations:
+            evaluation_start_datetime = course_end.replace(hour=8, minute=0) + timedelta(days=1)
```

Comment:
Maybe add `assert course_end.second == course_end.microsecond == 0` in case the input format changes to include seconds?